pkg/provision/workspace/deployment.go

```diff
@@ -214,9 +229,10 @@ func getSpecDeployment(
 					constants.DevWorkspaceIDLabel: workspace.Status.DevWorkspaceId,
 				},
 			},
-			Strategy: appsv1.DeploymentStrategy{
-				Type: appsv1.RecreateDeploymentStrategyType,
-			},
+			Strategy:                deploymentStrategy,
+			MinReadySeconds:         0,
+			RevisionHistoryLimit:    pointer.Int32(10),
+			ProgressDeadlineSeconds: pointer.Int32(600),
```

Comment:
I get that these are defaults but if we're setting them directly it may be worth considering whether these values are appropriate for our use case. 10 minutes for progress is a very long time, and we don't necessarily need 10 revisions in the history (and I believe this setting would result in up to ten replicasets for each deployment)

We should also verify what a failed rollout looks like, and if that's caught as a workspace failure or handled in some way by DWO.

Reply:
> if we're setting them directly it may be worth considering whether these values are appropriate for our use case. 10 minutes for progress is a very long time, and we don't necessarily need 10 revisions in the history (and I believe this setting would result in up to ten replicasets for each deployment)

Agreed. I initially added them because I thought they were necessary to prevent issues with the spec/cluster deployment diff, but they actually aren't. However, it's worth considering whether we should set (sane) values instead of letting them be the defaults set by Kubernetes. 

> We should also verify what a failed rollout looks like, and if that's caught as a workspace failure or handled in some way by DWO.

Agreed, will look into it.

Reply:
> We should also verify what a failed rollout looks like, and if that's caught as a workspace failure or handled in some way by DWO.

Not sure if this is sufficient of a check, but I checked the case you mentioned [here](https://github.com/eclipse/che/issues/21974#issuecomment-1453826209), and after modifying the memoryRequest and memoryLimit to 9Gi, the DevWorkspace failed:

```
[aobuchow@fedora devworkspace-operator]$ kubectl get dw -n $NAMESPACE -w
NAME                 DEVWORKSPACE ID             PHASE      INFO
code-latest          workspacefcb80b330a494679   Starting   Initializing DevWorkspace
code-latest          workspacefcb80b330a494679   Starting   Storage ready
code-latest          workspacefcb80b330a494679   Starting   Preparing networking
code-latest          workspacefcb80b330a494679   Starting   Preparing services
code-latest          workspacefcb80b330a494679   Starting   Preparing services
code-latest          workspacefcb80b330a494679   Starting   Networking ready
code-latest          workspacefcb80b330a494679   Starting   Waiting for DevWorkspace ServiceAccount
code-latest          workspacefcb80b330a494679   Starting   Waiting for DevWorkspace pull secrets
code-latest          workspacefcb80b330a494679   Starting   Waiting for workspace deployment
code-latest          workspacefcb80b330a494679   Starting   Waiting for editor to start
code-latest          workspacefcb80b330a494679   Running    https://workspacefcb80b330a494679.apps.ci-ln-ch9m0rb-72292.origin-ci-int-gce.dev.rhcloud.com/che-code/

code-latest          workspacefcb80b330a494679   Starting   Networking ready
code-latest          workspacefcb80b330a494679   Starting   Waiting for workspace deployment
code-latest          workspacefcb80b330a494679   Failing    Detected unrecoverable event FailedScheduling: 0/6 nodes are available: 1 Insufficient memory, 2 node(s) had volume node affinity conflict, 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }. preemption: 0/6 nodes are available: 1 No preemption victims found for incoming pod, 5 Preemption is not helpful for scheduling..
```

I still haven't tested getting the deployment rollout to not complete before  `ProgressDeadlineSeconds`

Reply:
> we don't necessarily need 10 revisions in the history (and I believe this setting would result in up to ten replicasets for each deployment)

Last thing I thought about for this PR: I checked the deployment spec and the default is indeed 10 for the revision history limit. Maybe this should be modified to 2? Or even 3? Or we can leave this for another PR/if an issue arises from this in the future.